plugin.cpp

```diff
@@ -248,6 +253,55 @@ PLUGIN_HANDLE plugin_init(ConfigCategory* config,
 	return (PLUGIN_HANDLE)info;
 }
 
+
+/**
+ * Flatten nested datapoint
+ *
+ * @param datapoint	datapoint to flatten
+ * @param assetName	assetname to which datapoint belongs
+ * @param newReadings	vector for newreadings of flatten datapoint
+ */
+void flattenDatapoint(Datapoint *datapoint, std::string assetName, vector<Reading *>& newReadings)
+{
+	static std::string datapointName = datapoint->getName();
```

Comment:
Why have you made this static? If it is to deal with the recursion and getting the base name right for the child and grandchild etc nodes then you are much better off by passing the base name as a argument with a default of an empty string. This way it will get restored on return correctly. I don't think what you have is going to work as you expect. In any case this static is something you should document in a comment as it is unusual coding practice.